src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { FC, useEffect, useMemo, useState } from 'react';
+import { Person } from '../types/Person';
+import { formatString } from '../servises/formatString';
+import debounce from 'lodash.debounce';
+
+type Props = {
+  people: Person[];
+  onSelect: (person: Person | null) => void;
+};
+
+export const Autocomplete: FC<Props> = ({ people, onSelect }) => {
+  const [query, setQuery] = useState('');
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useMemo(() => {
+    return debounce(setAppliedQuery, 300);
+  }, []);
+
+  useEffect(() => {
+    return () => applyQuery.cancel();
+  }, [applyQuery]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return people.filter(person =>
+      formatString(person.name).includes(appliedQuery),
+    );
+  }, [appliedQuery, people]);
+
+  return (
+    <>
+      <div className={`dropdown ${isDropdownVisible ? 'is-active' : ''}`}>
+        <div className="dropdown-trigger">
+          <input
+            type="text"
+            placeholder="Enter a part of the name"
+            className="input"
+            data-cy="search-input"
+            value={query}
+            onChange={handleQueryChange}
+            onFocus={() => {
+              setIsDropdownVisible(true);
+              onSelect(null);
+            }}
+          />
+        </div>
+
+        <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
+          <div className="dropdown-content">
+            {filteredPeople.map((person, index) => (
+              <div
+                key={index}
```

Comment:
Using the index as a key in a list can lead to issues if the list changes. Consider using a unique identifier from the `person` object, such as `person.id`, if available, to ensure stable keys.